app/src/main/java/com/a1ishka/bankaccount/presentation/account/AccountViewModel.kt

```diff
@@ -4,18 +4,32 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.a1ishka.bankaccount.data.repository.AccountRepositoryImpl
 import com.a1ishka.bankaccount.domain.Account
+import com.a1ishka.bankaccount.domain.repository.AccountRepository
+import com.a1ishka.bankaccount.util.toAccount
 import com.a1ishka.bankaccount.util.toAccountEntity
+import com.a1ishka.bankaccount.util.toAccountList
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+@HiltViewModel
 class AccountViewModel @Inject constructor(
-    private val accountRepositoryImpl: AccountRepositoryImpl
+    private val accountRepository: AccountRepository
 ) : ViewModel() {
     private val _state = MutableStateFlow(AccountState())
     val accountState: MutableStateFlow<AccountState> = _state
```

Comment:
must be StateFlow type instead of MutableStateFlow there. This method (when we have for example private _state - mutable and public state - immutable) called private setter and need to avoid state changes directly through the property